docker/puppetserver-standalone/docker-entrypoint.d/80-ca.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+
+master_running() {
+  status=$(curl --silent --fail --insecure "https://${CA_HOSTNAME}:8140/status/v1/simple")
+  test "$status" = "running"
+}
+
+hocon() {
+  /opt/puppetlabs/puppet/lib/ruby/vendor_gems/bin/hocon "$@"
+}
+
+CA_HOSTNAME="${CA_HOSTNAME:-puppet}"
+
+if [[ "$ENABLE_CA" != "true" ]]; then
+  # we are just an ordinary compiler
+  echo "turning off CA"
+  cat > /etc/puppetlabs/puppetserver/services.d/ca.cfg <<EOF
+puppetlabs.services.ca.certificate-authority-disabled-service/certificate-authority-disabled-service
+puppetlabs.trapperkeeper.services.watcher.filesystem-watch-service/filesystem-watch-service
+EOF
+
+  ssl_cert=$(puppet master --configprint hostcert)
+  ssl_key=$(puppet master --configprint hostprivkey)
+  ssl_ca_cert=$(puppet master --configprint localcacert)
+  certname=$(puppet config print certname)
```

Comment:
I'm not sure we should even bother with calling into `puppet config` or `puppet master --configprint` since we're assuming the filepaths anyway:
- https://github.com/puppetlabs/puppetserver/pull/1956/files#diff-f9e912c5472d58a4ee06a97dc9592516R33
- https://github.com/puppetlabs/puppetserver/pull/1956/files#diff-f9e912c5472d58a4ee06a97dc9592516R37

And we're using `$HOSTNAME` in place of `puppet config print certname` in other container-only places too.

Reply:
I think using `puppet config print certname` might be leftover from David's work where the compile masters and CA shared certs and certname != $HOSTNAME in that case, but I can probably get rid of that.

I think for the others I'd rather fix the places where the cert paths were assumed just to be more forward-proof / less fragile.

Reply:
mm, so it looks like if the `$PUPPETSERVER_HOSTNAME` env variable is set certname may not be the same as `$HOSTNAME`, so leaving that check in would probably be better

Reply:
> I think for the others I'd rather fix the places where the cert paths were assumed just to be more forward-proof / less fragile.

We can't just do this across the board because many containers won't have a puppet CLI to call into; puppetserver is kinda special in this case.

> mm, so it looks like if the $PUPPETSERVER_HOSTNAME env variable is set certname may not be the same as $HOSTNAME, so leaving that check in would probably be better

I wouldn't expect those two ENV variables to be the same, but that's not what I was talking about.

I was just saying that we've simplified calls like `puppet config print certname` into just `$HOSTNAME` elsewhere, and was thinking we could do that here to.

Would that not work here?

Not a problem if we leave it like this, since this is the one container that actually does have the `puppet config` CLI available.

Reply:
Oh I see how you updated this to not need the certname at all.

Nice!